test/testdrive/func.td

```diff
@@ -0,0 +1,42 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Test SQL functions.
+
+> SELECT split_part('abc~@~def~@~ghi', '~@~', 2);
+split_part
+---------
+def
+
+> SELECT split_part('abc~@~def~@~ghi', '', 1);
+split_part
+---------
+abc~@~def~@~ghi
+
+> SELECT split_part('abc~@~def~@~ghi', '~@~', 4);
+split_part
+----------
+""
+
+> SELECT split_part('hello there', '', 1);
```

Comment:
Would also be great to test the reverse case, where the string is empty but the delimiter is not.